source/common/access_log/access_log_formatter.cc

```diff
@@ -23,6 +23,9 @@ static const std::string UnspecifiedValueString = "-";
 
 namespace {
 
+// Matches newline pattern in a StartTimeFormatter format string.
+static const std::regex NEWLINEPATTERN{"%[1-9]*n", std::regex::optimize};
```

Comment:
I don't know if this is sufficient; if you look at `man strftime`, there are additional flags between the `%` and the decimal width specifier:
```
An optional decimal width specifier may follow the (possibly absent) flag. 
```

Reply:
Oops, I thought they'd only apply in numeric / alphabetic result strings, but testing it out seems like they do apply. Adding the change, and test cases.